src/main/java/chess/dao/ResultReader.java

```diff
@@ -0,0 +1,52 @@
+package chess.dao;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class ResultReader implements AutoCloseable {
+
+    private static final String READ_RESULT_EXCEPTION_MESSAGE = "잘못된 방식으로 조회 결과를 작업하였습니다.";
+
+    private final ResultSet resultSet;
+    private final Connection connection;
+
+    public ResultReader(ResultSet resultSet, Connection connection) {
+        this.resultSet = resultSet;
+        this.connection = connection;
+    }
+
+    public boolean nextRow() {
+        try {
+            return resultSet.next();
+        } catch (SQLException e) {
+            throw new IllegalStateException(READ_RESULT_EXCEPTION_MESSAGE);
+        }
+    }
+
+    public String readStringAt(String columnLabel) {
+        try {
+            return resultSet.getString(columnLabel);
+        } catch (SQLException e) {
+            throw new IllegalStateException(READ_RESULT_EXCEPTION_MESSAGE);
+        }
+    }
+
+    public int readFirstColumnAndClose() {
+        try (connection) {
+            nextRow();
+            return resultSet.getInt(1);
+        } catch (SQLException e) {
+            throw new IllegalStateException(READ_RESULT_EXCEPTION_MESSAGE);
+        }
+    }
+
+    @Override
+    public void close() {
+        try {
+            connection.close();
+        } catch (SQLException e) {
+            throw new IllegalStateException(READ_RESULT_EXCEPTION_MESSAGE);
+        }
+    }
+}
```

Comment:
```suggestion
}

```
깃헙에서 왜 파일 마지막에 newlline을 넣으라고 경고를 띄울까요?

Reply:
"마지막 줄이 비어있어야 파일이 끝났다고 인식하는 프로그램들이 존재하기 때문"이 맞나요